core/drivers/cbmem_console.c

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2023, Linaro Limited
+ */
+
+#include <compiler.h>
+#include <console.h>
+#include <drivers/cbmem_console.h>
+#include <io.h>
+#include <keep.h>
+#include <kernel/dt.h>
+#include <libfdt.h>
+#include <mm/core_mmu.h>
+#include <util.h>
+
+#define CURSOR_MASK (BIT(28) - 1)
+#define OVERFLOW BIT(31)
+
+/*
+ * Structures describing coreboot's in-memory descriptor tables. See
+ * <coreboot>/src/commonlib/include/commonlib/coreboot_tables.h for
+ * canonical implementation.
+ */
+
+struct cb_header {
+	char signature[4];
+	uint32_t header_bytes;
+	uint32_t header_checksum;
+	uint32_t table_bytes;
+	uint32_t table_checksum;
+	uint32_t table_entries;
+};
+
+#define CB_TAG_CBMEM_CONSOLE 0x17
+
+struct cb_entry {
+	uint32_t tag;
+	uint32_t size;
+	uint64_t uint64;
```

Comment:
itsn't it confusing to have a field alomst named like a type?

Reply:
Changed to 'value'.